static/app/components/nav/useNavIndicator.tsx

```diff
@@ -0,0 +1,55 @@
+import type {ComponentProps, HTMLAttributes, PointerEvent} from 'react';
+import {useFocusWithin, useHover} from '@react-aria/interactions';
+import {mergeProps} from '@react-aria/utils';
+import {type motion, useMotionValue, useReducedMotion, useSpring} from 'framer-motion';
+
+export interface IndicatorResult {
+  containerProps: HTMLAttributes<HTMLUListElement>;
+  indicatorProps: ComponentProps<typeof motion.span>;
+}
+
+export function useNavIndicator(): IndicatorResult {
+  const prefersReducedMotion = useReducedMotion();
+  const ty = useMotionValue(0);
+  const y = useSpring(
+    ty,
+    prefersReducedMotion
+      ? {duration: 0}
+      : {
+          mass: 0.5,
+          damping: 15,
+          stiffness: 200,
+          restDelta: 0.05,
+        }
+  );
+  const {isHovered, hoverProps} = useHover({});
+  const {focusWithinProps} = useFocusWithin({});
```

Comment:
I expected these hooks to allow not passing anything, but they unfortunately require an options `object`, but it can be empty. The overhead should be quite negligible in practice.